templates/components/products/card.html

```diff
@@ -43,7 +43,7 @@
                             <a href="{{url}}" class="button button--small card-figcaption-button" data-product-id="{{id}}">{{lang 'products.choose_options'}}</a>
                         {{/if}}
                         {{#if pre_order}}
-                            <a href="{{url}}" class="button button--small card-figcaption-button">{{lang 'products.pre_order'}}</a>
+                            <a href="{{pre_order_add_to_cart_url}}" class="button button--small card-figcaption-button">{{lang 'products.pre_order'}}</a>
```

Comment:
yes it will be a new fields which will only be passed for pre-order. On product details page it is a form so it is not required on those pages. But on pages where it is a card it is necessary bcoz earlier it was taking users to product detail pages. It is a similar url construct to add to cart.